File: src/main/java/net/nodus/database/auth/RefreshToken.java
package net.nodus.database.auth;

import java.time.Instant;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import net.nodus.database.common.MutableDocument;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "refresh_tokens")
@Getter
@Setter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class RefreshToken extends MutableDocument {

    @Id
    private String id;

    @Indexed
    private String userAccountId;

    @Indexed(unique = true)
    private String tokenHash;

    private Instant expiresAt;
    private Instant revokedAt;

    public RefreshToken(String userAccountId, String tokenHash, Instant expiresAt) {
        this.userAccountId = userAccountId;
        this.tokenHash = tokenHash;
        this.expiresAt = expiresAt;
    }
}

File: src/main/kotlin/net/nodus/auth/RefreshToken.kt
package net.nodus.auth

import net.nodus.common.MutableDocument
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.index.Indexed
import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant

@Document(collection = "refresh_tokens")
class RefreshToken(
    @Id
    val id: String? = null,

    @Indexed
    val userAccountId: String,

    @Indexed(unique = true)
    val tokenHash: String,

    val expiresAt: Instant,

    var revokedAt: Instant? = null,
) : MutableDocument()
